<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Create an anglemania_object from a Seurat Object — create_anglemania_object • anglemania</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create an anglemania_object from a Seurat Object — create_anglemania_object"><meta name="description" content="Constructs an anglemania_object-class from a given
Seurat object. This includes extracting and processing
count matrices, filtering genes based on expression in a minimum number of
cells, and storing results along with dataset and batch information. It also
calculates weights for each dataset or batch based on the number of samples."><meta property="og:description" content="Constructs an anglemania_object-class from a given
Seurat object. This includes extracting and processing
count matrices, filtering genes based on expression in a minimum number of
cells, and storing results along with dataset and batch information. It also
calculates weights for each dataset or batch based on the number of samples."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">anglemania</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/anglemania_tutorial.html">anglemania_tutorial</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/BIMSBbioinfo/anglemania/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Create an anglemania_object from a Seurat Object</h1>
      <small class="dont-index">Source: <a href="https://github.com/BIMSBbioinfo/anglemania/blob/master/R/objects.R" class="external-link"><code>R/objects.R</code></a></small>
      <div class="d-none name"><code>create_anglemania_object.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Constructs an <code><a href="anglemania_object-class.html">anglemania_object-class</a></code> from a given
<code><a href="https://satijalab.org/seurat/reference/Seurat-package.html" class="external-link">Seurat</a></code> object. This includes extracting and processing
count matrices, filtering genes based on expression in a minimum number of
cells, and storing results along with dataset and batch information. It also
calculates weights for each dataset or batch based on the number of samples.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">create_anglemania_object</span><span class="op">(</span></span>
<span>  <span class="va">seurat_object</span>,</span>
<span>  dataset_key <span class="op">=</span> <span class="cn">NA_character_</span>,</span>
<span>  <span class="va">batch_key</span>,</span>
<span>  min_cells_per_gene <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-seurat-object">seurat_object<a class="anchor" aria-label="anchor" href="#arg-seurat-object"></a></dt>
<dd><p>A <code><a href="https://satijalab.org/seurat/reference/Seurat-package.html" class="external-link">Seurat</a></code> object containing
single-cell RNA-seq data.</p></dd>


<dt id="arg-dataset-key">dataset_key<a class="anchor" aria-label="anchor" href="#arg-dataset-key"></a></dt>
<dd><p>A character string indicating the column name in the
Seurat object metadata that identifies the dataset to which each cell
belongs. If <code>NA</code>, all cells are assumed to belong to the same
dataset.</p></dd>


<dt id="arg-batch-key">batch_key<a class="anchor" aria-label="anchor" href="#arg-batch-key"></a></dt>
<dd><p>A character string indicating the column name(s) in the
Seurat object metadata that identify the batch to which each cell belongs.</p></dd>


<dt id="arg-min-cells-per-gene">min_cells_per_gene<a class="anchor" aria-label="anchor" href="#arg-min-cells-per-gene"></a></dt>
<dd><p>A numeric value indicating the minimum number of
cells in which a gene must be expressed to be included in the analysis.
Default is <code>1</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>An <code><a href="anglemania_object-class.html">anglemania_object-class</a></code> containing:</p><dl><dt><code>matrix_list</code></dt>
<dd><p>A list of filtered count matrices for each unique
batch.</p></dd>

<dt><code>dataset_key</code></dt>
<dd><p>The dataset key used for splitting the Seurat
object.</p></dd>

<dt><code>batch_key</code></dt>
<dd><p>The batch key used for splitting the Seurat
object.</p></dd>

<dt><code>data_info</code></dt>
<dd><p>A data frame summarizing the number of samples per
dataset and their weights.</p></dd>

<dt><code>weights</code></dt>
<dd><p>A numeric vector of weights for each dataset or batch
based on the number of samples.</p></dd>

<dt><code>intersect_genes</code></dt>
<dd><p>A character vector of genes expressed in at
least the specified number of cells across all batches.</p></dd>

<dt><code>min_cells_per_gene</code></dt>
<dd><p>The minimum number of cells per gene
threshold used for filtering.</p></dd>


</dl></div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The function performs the following steps:</p><ol><li><p>Adds a unique batch key to the Seurat object's metadata using
<code><a href="anglemania_object-methods.html">add_unique_batch_key</a></code>.</p></li>
<li><p>Extracts count matrices for each batch.</p></li>
<li><p>Filters genes based on the <code>min_cells_per_gene</code> threshold.</p></li>
<li><p>Identifies intersected genes present across all batches.</p></li>
<li><p>Converts count matrices to <code><a href="https://privefl.github.io/bigstatsr/reference/FBM-class.html" class="external-link">FBM</a></code> objects.</p></li>
<li><p>Computes weights for each batch or dataset.</p></li>
</ol></div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="anglemania_object-class.html">anglemania_object-class</a></code>,
<code><a href="anglemania_object-methods.html">add_unique_batch_key</a></code>,
<code><a href="anglemania.html">anglemania</a></code>,
<code><a href="https://privefl.github.io/bigstatsr/reference/FBM-class.html" class="external-link">FBM</a></code></p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu">SeuratObject</span><span class="fu">::</span><span class="va"><a href="https://mojaveazure.github.io/seurat-object/reference/pbmc_small.html" class="external-link">pbmc_small</a></span></span></span>
<span class="r-in"><span><span class="va">angl</span> <span class="op">&lt;-</span> <span class="fu">create_anglemania_object</span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">se</span>,</span></span>
<span class="r-in"><span>  batch_key <span class="op">=</span> <span class="st">"groups"</span>,</span></span>
<span class="r-in"><span>  min_cells_per_gene <span class="op">=</span> <span class="fl">1</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No dataset_key specified.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Assuming that all samples belong to the same dataset and are separated by batch_key: groups</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Extracting count matrices...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering each batch to at least 1 cells per gene...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Using the intersection of filtered genes from all batches...</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Number of genes in intersected set: 228</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                  | 0 % elapsed=00s     |==================================================| 100% elapsed=00s, remaining~00s</span>
<span class="r-in"><span> <span class="va">angl</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> anglemania_object</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> --------------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Dataset key: NA </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Batch key: groups </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of datasets: 1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Total number of batches: 2 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Batches (showing first 5):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> g2, g1 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Number of intersected genes: 228 </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Intersected genes (showing first 10):</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> MS4A1, CD79B, CD79A, HLA-DRA, TCL1A, HLA-DQB1, HVCN1, HLA-DMB, LTB, LINC00926 , ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> Min cells per gene: 1 </span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Aaron Kollotzek, Vedran Franke, Artem Baranovskii, Altuna Akalin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

