<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>anglemania_tutorial â€¢ anglemania</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="anglemania_tutorial">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">anglemania</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/anglemania_tutorial.html">anglemania_tutorial</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/BIMSBbioinfo/anglemania/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>anglemania_tutorial</h1>
            
            <h4 data-toc-skip class="date">2024-11-26</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/BIMSBbioinfo/anglemania/blob/master/vignettes/anglemania_tutorial.Rmd" class="external-link"><code>vignettes/anglemania_tutorial.Rmd</code></a></small>
      <div class="d-none name"><code>anglemania_tutorial.Rmd</code></div>
    </div>

    
    
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioinformatics.mdc-berlin.de/anglemania" class="external-link">anglemania</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/splatter/" class="external-link">splatter</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="create-simulated-data">Create simulated data<a class="anchor" aria-label="anchor" href="#create-simulated-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">batch.facLoc</span> <span class="op">&lt;-</span> <span class="fl">0.4</span></span>
<span><span class="va">de.facLoc</span> <span class="op">&lt;-</span> <span class="fl">0.1</span></span>
<span><span class="va">nBatches</span> <span class="op">&lt;-</span> <span class="fl">4</span></span>
<span><span class="va">nGroups</span> <span class="op">&lt;-</span> <span class="fl">3</span></span>
<span><span class="va">nGenes</span> <span class="op">&lt;-</span> <span class="fl">5000</span></span>
<span><span class="va">groupCells</span> <span class="op">&lt;-</span> <span class="fl">800</span></span>
<span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="http://oshlack.github.io/splatter/reference/splatSimulate.html" class="external-link">splatSimulate</a></span><span class="op">(</span></span>
<span>    batchCells <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">300</span> <span class="op">*</span> <span class="va">nGroups</span>, <span class="va">nBatches</span><span class="op">)</span>,</span>
<span>    batch.facLoc <span class="op">=</span> <span class="va">batch.facLoc</span>,</span>
<span>    group.prob <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span><span class="op">/</span><span class="va">nGroups</span>, <span class="va">nGroups</span><span class="op">)</span>,</span>
<span>    nGenes <span class="op">=</span> <span class="va">nGenes</span>,</span>
<span>    batch.facScale <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    method <span class="op">=</span> <span class="st">"groups"</span>,</span>
<span>    verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>    out.prob    <span class="op">=</span> <span class="fl">0.001</span>,</span>
<span>    de.prob     <span class="op">=</span> <span class="fl">0.1</span>, <span class="co"># mild </span></span>
<span>    de.facLoc   <span class="op">=</span> <span class="va">de.facLoc</span>,</span>
<span>    de.facScale <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>    bcv.common  <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">se</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://mojaveazure.github.io/seurat-object/reference/CreateSeuratObject.html" class="external-link">CreateSeuratObject</a></span><span class="op">(</span>counts <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/dge.html" class="external-link">counts</a></span><span class="op">(</span><span class="va">sim</span><span class="op">)</span>, meta.data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu">colData</span><span class="op">(</span><span class="va">sim</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning: Data is of class matrix. Coercing to dgCMatrix.</span></span></code></pre>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/rm.html" class="external-link">rm</a></span><span class="op">(</span><span class="va">sim</span><span class="op">)</span></span>
<span><span class="va">se</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 5000 features across 3600 samples within 1 assay </span></span>
<span><span class="co">## Active assay: RNA (5000 features, 0 variable features)</span></span>
<span><span class="co">##  1 layer present: counts</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">se</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          orig.ident nCount_RNA nFeature_RNA  Cell  Batch  Group ExpLibSize</span></span>
<span><span class="co">## Cell1 SeuratProject      81008         4004 Cell1 Batch1 Group2   80853.00</span></span>
<span><span class="co">## Cell2 SeuratProject      43335         3625 Cell2 Batch1 Group3   43357.87</span></span>
<span><span class="co">## Cell3 SeuratProject      61149         3859 Cell3 Batch1 Group2   61126.45</span></span>
<span><span class="co">## Cell4 SeuratProject      53996         3763 Cell4 Batch1 Group1   53628.79</span></span>
<span><span class="co">## Cell5 SeuratProject      66404         3881 Cell5 Batch1 Group1   65474.87</span></span>
<span><span class="co">## Cell6 SeuratProject      57973         3842 Cell6 Batch1 Group3   57751.74</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="unintegrated-data">unintegrated data<a class="anchor" aria-label="anchor" href="#unintegrated-data"></a>
</h2>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_unintegrated</span> <span class="op">&lt;-</span> <span class="va">se</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">se_unintegrated</span> <span class="op">&lt;-</span> <span class="va">se_unintegrated</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/NormalizeData.html" class="external-link">NormalizeData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/FindVariableFeatures.html" class="external-link">FindVariableFeatures</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/ScaleData.html" class="external-link">ScaleData</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunPCA.html" class="external-link">RunPCA</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://satijalab.org/seurat/reference/RunUMAP.html" class="external-link">RunUMAP</a></span><span class="op">(</span>dims <span class="op">=</span> <span class="fl">1</span><span class="op">:</span><span class="fl">30</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Normalizing layer: counts</span></span></code></pre>
<pre><code><span><span class="co">## Finding variable features for layer counts</span></span></code></pre>
<pre><code><span><span class="co">## Centering and scaling data matrix</span></span></code></pre>
<pre><code><span><span class="co">## PC_ 1 </span></span>
<span><span class="co">## Positive:  Gene4998, Gene1851, Gene1858, Gene1769, Gene4167, Gene4359, Gene3114, Gene1392, Gene906, Gene2555 </span></span>
<span><span class="co">##     Gene1186, Gene2855, Gene4451, Gene329, Gene2316, Gene2668, Gene1276, Gene1192, Gene946, Gene3822 </span></span>
<span><span class="co">##     Gene35, Gene3183, Gene3668, Gene4055, Gene3778, Gene2967, Gene3604, Gene1666, Gene80, Gene2950 </span></span>
<span><span class="co">## Negative:  Gene1778, Gene4947, Gene3361, Gene2804, Gene3191, Gene3939, Gene3248, Gene2846, Gene4023, Gene2954 </span></span>
<span><span class="co">##     Gene670, Gene1749, Gene1520, Gene2384, Gene543, Gene2777, Gene4148, Gene2684, Gene4012, Gene2408 </span></span>
<span><span class="co">##     Gene4822, Gene805, Gene4522, Gene4760, Gene475, Gene4756, Gene4161, Gene4226, Gene2347, Gene736 </span></span>
<span><span class="co">## PC_ 2 </span></span>
<span><span class="co">## Positive:  Gene4898, Gene3827, Gene2969, Gene4182, Gene4812, Gene4914, Gene4247, Gene950, Gene811, Gene280 </span></span>
<span><span class="co">##     Gene4844, Gene1746, Gene3652, Gene3144, Gene4773, Gene4852, Gene1929, Gene1021, Gene2554, Gene3403 </span></span>
<span><span class="co">##     Gene1268, Gene254, Gene915, Gene938, Gene2843, Gene4489, Gene709, Gene4074, Gene4667, Gene2103 </span></span>
<span><span class="co">## Negative:  Gene4768, Gene2483, Gene2444, Gene1171, Gene448, Gene575, Gene3564, Gene2433, Gene351, Gene912 </span></span>
<span><span class="co">##     Gene1870, Gene3060, Gene2568, Gene4128, Gene1786, Gene2891, Gene4209, Gene4637, Gene297, Gene2010 </span></span>
<span><span class="co">##     Gene787, Gene492, Gene3687, Gene4718, Gene3262, Gene1657, Gene1493, Gene2031, Gene4101, Gene4740 </span></span>
<span><span class="co">## PC_ 3 </span></span>
<span><span class="co">## Positive:  Gene267, Gene1099, Gene1610, Gene505, Gene471, Gene4737, Gene4440, Gene4515, Gene4485, Gene3812 </span></span>
<span><span class="co">##     Gene2434, Gene1179, Gene2501, Gene3619, Gene4586, Gene4343, Gene4125, Gene4879, Gene3772, Gene4654 </span></span>
<span><span class="co">##     Gene3567, Gene3889, Gene1562, Gene19, Gene3077, Gene4830, Gene2132, Gene3466, Gene3294, Gene4287 </span></span>
<span><span class="co">## Negative:  Gene463, Gene3025, Gene2894, Gene444, Gene840, Gene95, Gene816, Gene4617, Gene3813, Gene4709 </span></span>
<span><span class="co">##     Gene101, Gene1699, Gene3385, Gene3369, Gene2122, Gene4376, Gene1006, Gene1685, Gene3846, Gene4490 </span></span>
<span><span class="co">##     Gene1590, Gene2521, Gene4858, Gene4463, Gene2239, Gene274, Gene4673, Gene289, Gene3068, Gene3290 </span></span>
<span><span class="co">## PC_ 4 </span></span>
<span><span class="co">## Positive:  Gene1312, Gene4793, Gene1488, Gene2650, Gene389, Gene1558, Gene1133, Gene793, Gene4895, Gene1582 </span></span>
<span><span class="co">##     Gene4231, Gene4586, Gene1782, Gene3570, Gene471, Gene554, Gene463, Gene2214, Gene1147, Gene2496 </span></span>
<span><span class="co">##     Gene3077, Gene4490, Gene2761, Gene3462, Gene3141, Gene2312, Gene3827, Gene9, Gene533, Gene4890 </span></span>
<span><span class="co">## Negative:  Gene345, Gene361, Gene670, Gene470, Gene4238, Gene4869, Gene2382, Gene4738, Gene4661, Gene816 </span></span>
<span><span class="co">##     Gene645, Gene2552, Gene2521, Gene128, Gene2842, Gene23, Gene4363, Gene4445, Gene2245, Gene1340 </span></span>
<span><span class="co">##     Gene3183, Gene1464, Gene1843, Gene3168, Gene564, Gene1419, Gene280, Gene4215, Gene4205, Gene2573 </span></span>
<span><span class="co">## PC_ 5 </span></span>
<span><span class="co">## Positive:  Gene816, Gene4403, Gene3168, Gene4299, Gene4969, Gene1651, Gene3618, Gene2512, Gene4616, Gene1419 </span></span>
<span><span class="co">##     Gene4350, Gene777, Gene4388, Gene634, Gene2354, Gene4509, Gene4231, Gene35, Gene3992, Gene2842 </span></span>
<span><span class="co">##     Gene2936, Gene2282, Gene3846, Gene4982, Gene1236, Gene3676, Gene1768, Gene350, Gene2434, Gene1659 </span></span>
<span><span class="co">## Negative:  Gene4586, Gene1851, Gene4797, Gene265, Gene4014, Gene920, Gene4574, Gene3377, Gene3570, Gene3141 </span></span>
<span><span class="co">##     Gene2985, Gene2687, Gene3077, Gene1052, Gene3567, Gene463, Gene584, Gene3328, Gene2860, Gene4407 </span></span>
<span><span class="co">##     Gene3493, Gene3228, Gene1647, Gene1160, Gene337, Gene1312, Gene1921, Gene3462, Gene3564, Gene4065</span></span></code></pre>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">se_unintegrated</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anglemania_tutorial_files/figure-html/unnamed-chunk-1-1.png" width="700"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">se_unintegrated</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anglemania_tutorial_files/figure-html/unnamed-chunk-1-2.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="run-anglemania">run anglemania<a class="anchor" aria-label="anchor" href="#run-anglemania"></a>
</h2>
<div class="section level3">
<h3 id="create-anglem-object">create anglem object<a class="anchor" aria-label="anchor" href="#create-anglem-object"></a>
</h3>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">se</span><span class="op">[[</span><span class="op">]</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          orig.ident nCount_RNA nFeature_RNA  Cell  Batch  Group ExpLibSize</span></span>
<span><span class="co">## Cell1 SeuratProject      81008         4004 Cell1 Batch1 Group2   80853.00</span></span>
<span><span class="co">## Cell2 SeuratProject      43335         3625 Cell2 Batch1 Group3   43357.87</span></span>
<span><span class="co">## Cell3 SeuratProject      61149         3859 Cell3 Batch1 Group2   61126.45</span></span>
<span><span class="co">## Cell4 SeuratProject      53996         3763 Cell4 Batch1 Group1   53628.79</span></span>
<span><span class="co">## Cell5 SeuratProject      66404         3881 Cell5 Batch1 Group1   65474.87</span></span>
<span><span class="co">## Cell6 SeuratProject      57973         3842 Cell6 Batch1 Group3   57751.74</span></span></code></pre>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">batch_key</span> <span class="op">&lt;-</span> <span class="st">"Batch"</span></span>
<span></span>
<span><span class="va">angl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_anglemania_object.html">create_anglemania_object</a></span><span class="op">(</span><span class="va">se</span>,</span>
<span>  batch_key <span class="op">=</span> <span class="va">batch_key</span>,</span>
<span>  min_cells_per_gene <span class="op">=</span> <span class="fl">1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## No dataset_key specified.</span></span>
<span><span class="co">## Assuming that all samples belong to the same dataset and are separated by batch_key: Batch</span></span></code></pre>
<pre><code><span><span class="co">## Extracting count matrices...</span></span></code></pre>
<pre><code><span><span class="co">## Filtering each batch to at least 1 cells per gene...</span></span></code></pre>
<pre><code><span><span class="co">## Using the intersection of filtered genes from all batches...</span></span></code></pre>
<pre><code><span><span class="co">## Number of genes in intersected set: 4958</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">angl</span></span></code></pre></div>
<pre><code><span><span class="co">## anglemania_object</span></span>
<span><span class="co">## --------------</span></span>
<span><span class="co">## Dataset key: NA </span></span>
<span><span class="co">## Batch key: Batch </span></span>
<span><span class="co">## Number of datasets: 1 </span></span>
<span><span class="co">## Total number of batches: 4 </span></span>
<span><span class="co">## Batches (showing first 5):</span></span>
<span><span class="co">## Batch1, Batch2, Batch3, Batch4 </span></span>
<span><span class="co">## Number of intersected genes: 4958 </span></span>
<span><span class="co">## Intersected genes (showing first 10):</span></span>
<span><span class="co">## Gene1, Gene2, Gene3, Gene4, Gene5, Gene6, Gene7, Gene8, Gene9, Gene10 , ...</span></span>
<span><span class="co">## Min cells per gene: 1</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="run-anglemania-1">run anglemania<a class="anchor" aria-label="anchor" href="#run-anglemania-1"></a>
</h3>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">angl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/anglemania.html">anglemania</a></span><span class="op">(</span><span class="va">angl</span>,</span>
<span>  zscore_mean_threshold <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  zscore_sn_threshold <span class="op">=</span> <span class="fl">2</span>,</span>
<span>  max_n_genes <span class="op">=</span> <span class="fl">2000</span> <span class="co"># optionally define a max number of genes. default is 2000</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Computing angles and transforming to z-scores...</span></span></code></pre>
<pre><code><span><span class="co">## Computing statistics...</span></span></code></pre>
<pre><code><span><span class="co">## Weighting matrix_list...</span></span></code></pre>
<pre><code><span><span class="co">## Calculating mean...</span></span></code></pre>
<pre><code><span><span class="co">## Calculating sds...</span></span></code></pre>
<pre><code><span><span class="co">## Filtering features...</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">angl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/select_genes.html">select_genes</a></span><span class="op">(</span><span class="va">angl</span>,</span>
<span>                     zscore_mean_threshold <span class="op">=</span> <span class="fl">2.5</span>,</span>
<span>                     zscore_sn_threshold <span class="op">=</span> <span class="fl">2.5</span><span class="op">)</span></span>
<span><span class="co"># Inspect the anglemania genes</span></span>
<span><span class="va">integration_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/anglemania_object-methods.html">get_anglemania_genes</a></span><span class="op">(</span><span class="va">angl</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">integration_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Gene816"  "Gene4124" "Gene4586" "Gene4679" "Gene1921" "Gene680"</span></span></code></pre>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">integration_genes</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] 291</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="integration">integration<a class="anchor" aria-label="anchor" href="#integration"></a>
</h3>
<ul>
<li>we implemented the easy-to-use integrate_by_features() function from
the anglemania package which uses Seurat CCA integration</li>
<li>you can also just use the anglemania genes for other integration
algorithms</li>
</ul>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># if the Seurat FindIntegrationAnchors() function does not work,</span></span>
<span><span class="co"># change this to the specified size:</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span>future.globals.maxSize <span class="op">=</span> <span class="fl">4000</span> <span class="op">*</span> <span class="fl">1024</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">seurat_integrated_angl</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integrate_by_features.html">integrate_by_features</a></span><span class="op">(</span><span class="va">se</span>,</span>
<span>    <span class="va">angl</span>,</span>
<span>    process <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Log normalizing data...</span></span></code></pre>
<pre><code><span><span class="co">## Finding integration anchors...</span></span></code></pre>
<pre><code><span><span class="co">## Integrating samples...</span></span></code></pre>
<pre><code><span><span class="co">## Running PCA with 30 PCs</span></span></code></pre>
<pre><code><span><span class="co">## Running UMAP with 30 PCs and 30 neighbors</span></span></code></pre>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_integrated_angl</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 5291 features across 3600 samples within 2 assays </span></span>
<span><span class="co">## Active assay: integrated (291 features, 291 variable features)</span></span>
<span><span class="co">##  2 layers present: data, scale.data</span></span>
<span><span class="co">##  1 other assay present: RNA</span></span>
<span><span class="co">##  2 dimensional reductions calculated: pca, umap</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="plot-integration">plot integration<a class="anchor" aria-label="anchor" href="#plot-integration"></a>
</h3>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_integrated_angl</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anglemania_tutorial_files/figure-html/unnamed-chunk-2-1.png" width="700"></p>
<div class="sourceCode" id="cb45"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_integrated_angl</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anglemania_tutorial_files/figure-html/unnamed-chunk-2-2.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="comparison-to-using-hvgs">Comparison to using HVGs<a class="anchor" aria-label="anchor" href="#comparison-to-using-hvgs"></a>
</h2>
<div class="section level3">
<h3 id="get-hvgs">get HVGs<a class="anchor" aria-label="anchor" href="#get-hvgs"></a>
</h3>
<div class="sourceCode" id="cb46"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">se_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://satijalab.org/seurat/reference/SplitObject.html" class="external-link">SplitObject</a></span><span class="op">(</span><span class="va">se</span>, split.by <span class="op">=</span> <span class="va">batch_key</span><span class="op">)</span></span>
<span><span class="va">se_list</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">se_list</span>, <span class="va">NormalizeData</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Normalizing layer: counts</span></span>
<span><span class="co">## Normalizing layer: counts</span></span>
<span><span class="co">## Normalizing layer: counts</span></span>
<span><span class="co">## Normalizing layer: counts</span></span></code></pre>
<div class="sourceCode" id="cb48"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># anglemania chose only 291 genes based on the thresholds</span></span>
<span><span class="co">#   to make it comparable we only choose 300 highly variable genes:</span></span>
<span><span class="va">hvg_features</span> <span class="op">&lt;-</span> <span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/SelectIntegrationFeatures.html" class="external-link">SelectIntegrationFeatures</a></span><span class="op">(</span><span class="va">se_list</span>, nfeatures <span class="op">=</span> <span class="fl">300</span><span class="op">)</span> </span></code></pre></div>
<pre><code><span><span class="co">## No variable features found for object1 in the object.list. Running FindVariableFeatures ...</span></span></code></pre>
<pre><code><span><span class="co">## Finding variable features for layer counts</span></span></code></pre>
<pre><code><span><span class="co">## No variable features found for object2 in the object.list. Running FindVariableFeatures ...</span></span></code></pre>
<pre><code><span><span class="co">## Finding variable features for layer counts</span></span></code></pre>
<pre><code><span><span class="co">## No variable features found for object3 in the object.list. Running FindVariableFeatures ...</span></span></code></pre>
<pre><code><span><span class="co">## Finding variable features for layer counts</span></span></code></pre>
<pre><code><span><span class="co">## No variable features found for object4 in the object.list. Running FindVariableFeatures ...</span></span></code></pre>
<pre><code><span><span class="co">## Finding variable features for layer counts</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="integration-1">integration<a class="anchor" aria-label="anchor" href="#integration-1"></a>
</h3>
<div class="sourceCode" id="cb57"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/warning.html" class="external-link">suppressWarnings</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="va">seurat_integrated_hvg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/integrate_seurat_list.html">integrate_seurat_list</a></span><span class="op">(</span><span class="va">se_list</span>,</span>
<span>    features <span class="op">=</span> <span class="va">hvg_features</span>,</span>
<span>    process <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Log normalizing data...</span></span></code></pre>
<pre><code><span><span class="co">## Finding integration anchors...</span></span></code></pre>
<pre><code><span><span class="co">## Integrating samples...</span></span></code></pre>
<pre><code><span><span class="co">## Running PCA with 30 PCs</span></span></code></pre>
<pre><code><span><span class="co">## Running UMAP with 30 PCs and 30 neighbors</span></span></code></pre>
<div class="sourceCode" id="cb63"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">seurat_integrated_hvg</span></span></code></pre></div>
<pre><code><span><span class="co">## An object of class Seurat </span></span>
<span><span class="co">## 5300 features across 3600 samples within 2 assays </span></span>
<span><span class="co">## Active assay: integrated (300 features, 300 variable features)</span></span>
<span><span class="co">##  2 layers present: data, scale.data</span></span>
<span><span class="co">##  1 other assay present: RNA</span></span>
<span><span class="co">##  2 dimensional reductions calculated: pca, umap</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="plot-integration-1">plot integration<a class="anchor" aria-label="anchor" href="#plot-integration-1"></a>
</h3>
<div class="sourceCode" id="cb65"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_integrated_hvg</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Batch"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anglemania_tutorial_files/figure-html/unnamed-chunk-3-1.png" width="700"></p>
<div class="sourceCode" id="cb66"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">Seurat</span><span class="fu">::</span><span class="fu"><a href="https://satijalab.org/seurat/reference/DimPlot.html" class="external-link">DimPlot</a></span><span class="op">(</span><span class="va">seurat_integrated_hvg</span>, reduction <span class="op">=</span> <span class="st">"umap"</span>, group.by <span class="op">=</span> <span class="st">"Group"</span><span class="op">)</span></span></code></pre></div>
<p><img src="anglemania_tutorial_files/figure-html/unnamed-chunk-3-2.png" width="700"></p>
<div class="sourceCode" id="cb67"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.2 (2024-10-31)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.5 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] splatter_1.30.0             SingleCellExperiment_1.28.1</span></span>
<span><span class="co">##  [3] SummarizedExperiment_1.36.0 Biobase_2.66.0             </span></span>
<span><span class="co">##  [5] GenomicRanges_1.58.0        GenomeInfoDb_1.42.0        </span></span>
<span><span class="co">##  [7] IRanges_2.40.0              S4Vectors_0.44.0           </span></span>
<span><span class="co">##  [9] BiocGenerics_0.52.0         MatrixGenerics_1.18.0      </span></span>
<span><span class="co">## [11] matrixStats_1.4.1           Seurat_5.1.0               </span></span>
<span><span class="co">## [13] SeuratObject_5.0.2          sp_2.1-4                   </span></span>
<span><span class="co">## [15] dplyr_1.1.4                 anglemania_0.99.0          </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] RcppAnnoy_0.0.22        splines_4.4.2           later_1.3.2            </span></span>
<span><span class="co">##   [4] tibble_3.2.1            polyclip_1.10-7         fastDummies_1.7.4      </span></span>
<span><span class="co">##   [7] lifecycle_1.0.4         doParallel_1.0.17       globals_0.16.3         </span></span>
<span><span class="co">##  [10] lattice_0.22-6          MASS_7.3-61             backports_1.5.0        </span></span>
<span><span class="co">##  [13] magrittr_2.0.3          plotly_4.10.4           sass_0.4.9             </span></span>
<span><span class="co">##  [16] rmarkdown_2.29          jquerylib_0.1.4         yaml_2.3.10            </span></span>
<span><span class="co">##  [19] bigparallelr_0.3.2      httpuv_1.6.15           sctransform_0.4.1      </span></span>
<span><span class="co">##  [22] spam_2.11-0             spatstat.sparse_3.1-0   reticulate_1.40.0      </span></span>
<span><span class="co">##  [25] cowplot_1.1.3           pbapply_1.7-2           RColorBrewer_1.1-3     </span></span>
<span><span class="co">##  [28] abind_1.4-8             zlibbioc_1.52.0         Rtsne_0.17             </span></span>
<span><span class="co">##  [31] purrr_1.0.2             bigassertr_0.1.6        GenomeInfoDbData_1.2.13</span></span>
<span><span class="co">##  [34] ggrepel_0.9.6           irlba_2.3.5.1           listenv_0.9.1          </span></span>
<span><span class="co">##  [37] spatstat.utils_3.1-1    goftest_1.2-3           RSpectra_0.16-2        </span></span>
<span><span class="co">##  [40] spatstat.random_3.3-2   fitdistrplus_1.2-1      parallelly_1.39.0      </span></span>
<span><span class="co">##  [43] pkgdown_2.1.1           leiden_0.4.3.1          codetools_0.2-20       </span></span>
<span><span class="co">##  [46] DelayedArray_0.32.0     tidyselect_1.2.1        UCSC.utils_1.2.0       </span></span>
<span><span class="co">##  [49] farver_2.1.2            bigstatsr_1.6.1         spatstat.explore_3.3-3 </span></span>
<span><span class="co">##  [52] flock_0.7               jsonlite_1.8.9          progressr_0.15.1       </span></span>
<span><span class="co">##  [55] ggridges_0.5.6          survival_3.7-0          iterators_1.0.14       </span></span>
<span><span class="co">##  [58] systemfonts_1.1.0       foreach_1.5.2           tools_4.4.2            </span></span>
<span><span class="co">##  [61] ragg_1.3.3              ica_1.0-3               Rcpp_1.0.13-1          </span></span>
<span><span class="co">##  [64] glue_1.8.0              gridExtra_2.3           SparseArray_1.6.0      </span></span>
<span><span class="co">##  [67] xfun_0.49               withr_3.0.2             fastmap_1.2.0          </span></span>
<span><span class="co">##  [70] fansi_1.0.6             digest_0.6.37           R6_2.5.1               </span></span>
<span><span class="co">##  [73] mime_0.12               textshaping_0.4.0       colorspace_2.1-1       </span></span>
<span><span class="co">##  [76] scattermore_1.2         tensor_1.5              spatstat.data_3.1-4    </span></span>
<span><span class="co">##  [79] utf8_1.2.4              tidyr_1.3.1             generics_0.1.3         </span></span>
<span><span class="co">##  [82] data.table_1.16.2       httr_1.4.7              htmlwidgets_1.6.4      </span></span>
<span><span class="co">##  [85] S4Arrays_1.6.0          uwot_0.2.2              pkgconfig_2.0.3        </span></span>
<span><span class="co">##  [88] gtable_0.3.6            lmtest_0.9-40           XVector_0.46.0         </span></span>
<span><span class="co">##  [91] htmltools_0.5.8.1       dotCall64_1.2           scales_1.3.0           </span></span>
<span><span class="co">##  [94] png_0.1-8               spatstat.univar_3.1-1   knitr_1.49             </span></span>
<span><span class="co">##  [97] reshape2_1.4.4          checkmate_2.3.2         nlme_3.1-166           </span></span>
<span><span class="co">## [100] cachem_1.1.0            zoo_1.8-12              stringr_1.5.1          </span></span>
<span><span class="co">## [103] rmio_0.4.0              KernSmooth_2.23-24      parallel_4.4.2         </span></span>
<span><span class="co">## [106] miniUI_0.1.1.1          desc_1.4.3              pillar_1.9.0           </span></span>
<span><span class="co">## [109] grid_4.4.2              vctrs_0.6.5             RANN_2.6.2             </span></span>
<span><span class="co">## [112] promises_1.3.0          ff_4.5.0                xtable_1.8-4           </span></span>
<span><span class="co">## [115] cluster_2.1.6           evaluate_1.0.1          cli_3.6.3              </span></span>
<span><span class="co">## [118] locfit_1.5-9.10         compiler_4.4.2          rlang_1.1.4            </span></span>
<span><span class="co">## [121] crayon_1.5.3            future.apply_1.11.3     labeling_0.4.3         </span></span>
<span><span class="co">## [124] ps_1.8.1                plyr_1.8.9              fs_1.6.5               </span></span>
<span><span class="co">## [127] stringi_1.8.4           viridisLite_0.4.2       deldir_2.0-4           </span></span>
<span><span class="co">## [130] BiocParallel_1.40.0     munsell_0.5.1           lazyeval_0.2.2         </span></span>
<span><span class="co">## [133] spatstat.geom_3.3-4     Matrix_1.7-1            RcppHNSW_0.6.0         </span></span>
<span><span class="co">## [136] patchwork_1.3.0         future_1.34.0           ggplot2_3.5.1          </span></span>
<span><span class="co">## [139] shiny_1.9.1             ROCR_1.0-11             igraph_2.1.1           </span></span>
<span><span class="co">## [142] bslib_0.8.0             bit_4.5.0</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Aaron Kollotzek, Vedran Franke, Artem Baranovskii, Altuna Akalin.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
